File: project_euler/p013/solution.hs
{--
Work out the first ten digits of the sum of the following one-hundred 50-digit numbers.

(see data.txt)
--}

module Main where

import System.IO

parse :: String -> [Integer]
parse input = map (read) (lines input)


main :: IO ()
main = do
    fd <- openFile "data.txt" ReadMode
    contents <- hGetContents fd
    print (sum (parse contents))

File: project_euler/p013/solution.rs


fn main() {

    //Import data
    let numbers = include_str!("data.txt");

    let sum: String = numbers
        .lines()                            //Split by line
        .map(|x| x.parse::<f64>().unwrap()) //Convert to f64
        .sum::<f64>()                       //Sum
        .to_string()                        //Convert sum to string
        .chars()                            //Convert to iterable of chars
        .take(10)                           //Take the first 10
        .collect::<String>();               //Convert to a string again

    println!("Answer: {}", sum);
}

